src/pymmcore_plus/mda/metadata/schema.py

```diff
@@ -0,0 +1,169 @@
+    parent_label: Optional[str]  # None on hub devices
+
+    # state device only
+    labels: NotRequired[Tuple[str, ...]]
+    # focus device only
+    focus_direction: NotRequired[Literal["Unknown", "TowardSample", "AwayFromSample"]]
+    # hub device only
+    child_names: NotRequired[Tuple[str, ...]]
+    # stage or XYstage device only
+    is_stage_sequenceable: NotRequired[bool]
+    # stage device only
+    is_continuous_focus_drive: NotRequired[bool]
+    is_stage_linear_sequenceable: NotRequired[bool]
+    # camera device only
+    is_exposure_sequenceable: NotRequired[bool]
+
+
+class SystemInfo(TypedDict):
+    """General system information."""
+
+    pymmcore_version: str
+    pymmcore_plus_version: str
+    mmcore_version: str
+    device_api_version: str
+    device_adapter_search_paths: Tuple[str, ...]
+    system_configuration: Optional[str]
+    primary_log_file: str
+    circular_buffer_memory_footprint: int
+    buffer_total_capacity: int
+    buffer_free_capacity: int
+    timeout_ms: int
+    # remaining_image_count: int
+    continuous_focus_enabled: bool
+    continuous_focus_locked: bool
+    auto_shutter: bool
+
+
+class ImageInfo(TypedDict):
+    """Information about the current image structure."""
+
+    bytes_per_pixel: int
+    current_pixel_size_config: str
+    exposure: float
+    image_bit_depth: int
+    image_buffer_size: int
+    image_height: int
+    image_width: int
+    magnification_factor: float
+    number_of_camera_channels: int
+    number_of_components: int
+    pixel_size_affine: Tuple[float, float, float, float, float, float]
+    pixel_size_um: float
+    roi: List[int]
+    camera_device: str
+    multi_roi: NotRequired[Tuple[List[int], List[int], List[int], List[int]]]
+
+
+class Position(TypedDict):
+    """Represents a position in 3D space and focus."""
+
+    x: Optional[float]
+    y: Optional[float]
+    focus: Optional[float]
+
+
+class PropertyValue(TypedDict):
+    """A single device property setting in a configuration group."""
+
+    dev: str
+    prop: str
+    val: Any
+
+
+class ConfigPreset(TypedDict):
+    """A group of device property settings."""
+
+    name: str
+    settings: Tuple[PropertyValue, ...]
+
+
+class PixelSizeConfigPreset(ConfigPreset):
+    """A specialized group of device property settings for a pixel size preset."""
+
+    pixel_size_um: float
+    pixel_size_affine: Tuple[float, float, float, float, float, float]
+
+
+class ConfigGroup(TypedDict):
+    """A group of configuration presets."""
+
+    name: str
+    presets: Tuple[ConfigPreset, ...]
+
+
+class SummaryMetaV1(TypedDict, total=False):
+    """Complete summary metadata for the system. Version 1.0."""
+
+    devices: Tuple[DeviceInfo, ...]
+    system_info: SystemInfo
+    image_info: ImageInfo
+    config_groups: Tuple[ConfigGroup, ...]
+    pixel_size_configs: Tuple[PixelSizeConfigPreset, ...]
+    position: Position
+    mda_sequence: NotRequired[useq.MDASequence]
+    date_time: str
```

Comment:
Consider specifying that it must be UTC.

Reply:
what about a timezone aware timestamp, that doesn't necessarily have to be UTC offset 0:

```python
In [10]: datetime.datetime.now(datetime.UTC).astimezone().isoformat()
Out[10]: '2024-06-26T19:35:09.963496-04:00'

In [11]: datetime.datetime.now(datetime.UTC).isoformat()
Out[11]: '2024-06-26T23:35:18.202662+00:00'
```

Reply:
That should be fine since it has a bijection to/from UTC. Nice idea for human-friendliness!